components/ws-manager/pkg/manager/annotations.go

```diff
@@ -171,6 +168,32 @@ type workspaceDisposalStatus struct {
 	BackupComplete bool             `json:"backupComplete,omitempty"`
 	BackupFailure  string           `json:"backupFailure,omitempty"`
 	GitStatus      *csapi.GitStatus `json:"gitStatus,omitempty"`
+	Status         DisposalStatus   `json:"status,omitempty"`
+}
+
+func (m *Manager) markDisposalStatus(ctx context.Context, workspaceID string, disposalStatus *workspaceDisposalStatus) error {
+	b, err := json.Marshal(disposalStatus)
+	if err != nil {
+		return err
+	}
+
+	err = m.markWorkspace(ctx, workspaceID, addMark(disposalStatusAnnotation, string(b)))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type DisposalStatus string
+
+const (
```

Comment:
Sorry, may I ask you to give me more explanation?

Reply:
I think JenTing meant that if instead of doing this:
`ds == ""` you could instead replace it with 
`ds == DisposalEmpty` which you would to const above as
`DisposalEmpty DisposalStatus = ""`

Reply:
Correct, thanks Pavel's explanation 🙏